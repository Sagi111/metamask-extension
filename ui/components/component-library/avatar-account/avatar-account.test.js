/* eslint-disable jest/require-top-level-describe */
import { render } from '@testing-library/react';
import React from 'react';
import { AvatarAccount, AvatarAccountSize } from '.';
import 'jest-canvas-mock';

describe('AvatarAccount', () => {
  const args = {
    address: '0x5CfE73b6021E818B776b421B1c4Db2474086a7e1',
    variant: 'Jazzicon',
  };
  it('should render correctly', () => {
    const { getByTestId, container } = render(
      <AvatarAccount data-testid="avatar-account" {...args} />,
    );
    expect(getByTestId('avatar-account')).toBeDefined();
    expect(container).toMatchSnapshot();
  });

  it('should render Jazzicon correctly', () => {
    const container = (
      <AvatarAccount
        data-testid="avatar-account"
        {...args}
        variant="Jazzicon"
      />
    );
    expect(container.props.variant).toStrictEqual('Jazzicon');
  });

  it('should render Blockie correctly', () => {
    const container = (
      <AvatarAccount data-testid="avatar-account" {...args} variant="Blockie" />
    );
    expect(container.props.variant).toStrictEqual('Blockie');
  });

  it('should render with custom classname', () => {
    const { getByTestId } = render(
      <AvatarAccount
        className="mm-avatar-account--test"
        data-testid="classname"
        {...args}
      />,
    );
    expect(getByTestId('classname')).toHaveClass('mm-avatar-account--test');
  });

  it('should render with address', () => {
    const container = (
      <AvatarAccount
        className="mm-avatar-account--test"
        data-testid="classname"
        {...args}
        address="0x0"
      />
    );
    expect(container.props.address).toStrictEqual('0x0');
  });

  it('should render with different size classes', () => {
    const { getByTestId } = render(
      <>
        <AvatarAccount
          size={AvatarAccountSize.Xs}
          data-testid={AvatarAccountSize.Xs}
          {...args}
        />
        <AvatarAccount
          size={AvatarAccountSize.Sm}
          data-testid={AvatarAccountSize.Sm}
          {...args}
        />
        <AvatarAccount
          size={AvatarAccountSize.Md}
          data-testid={AvatarAccountSize.Md}
          {...args}
        />
        <AvatarAccount
          size={AvatarAccountSize.Lg}
          data-testid={AvatarAccountSize.Lg}
          {...args}
        />
        <AvatarAccount
          size={AvatarAccountSize.Xl}
          data-testid={AvatarAccountSize.Xl}
          {...args}
        />
      </>,
    );
    expect(getByTestId(AvatarAccountSize.Xs)).toHaveClass(
      `mm-avatar-base--size-${AvatarAccountSize.Xs}`,
    );
    expect(getByTestId(AvatarAccountSize.Sm)).toHaveClass(
      `mm-avatar-base--size-${AvatarAccountSize.Sm}`,
    );
    expect(getByTestId(AvatarAccountSize.Md)).toHaveClass(
      `mm-avatar-base--size-${AvatarAccountSize.Md}`,
    );
    expect(getByTestId(AvatarAccountSize.Lg)).toHaveClass(
      `mm-avatar-base--size-${AvatarAccountSize.Lg}`,
    );
    expect(getByTestId(AvatarAccountSize.Xl)).toHaveClass(
      `mm-avatar-base--size-${AvatarAccountSize.Xl}`,
    );
  });
});
