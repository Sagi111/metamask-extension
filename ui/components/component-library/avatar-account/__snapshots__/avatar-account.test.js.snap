// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarAccount should render correctly 1`] = `
<div>
  <div
    class="box mm-avatar-base mm-avatar-base--size-md mm-avatar-account box--flex-direction-row box--color-text-default box--background-color-background-alternative box--border-color-border-default box--border-style-solid box--border-width-1"
    data-testid="avatar-account"
  >
    <div
      class="mm-avatar-account__jazzicon"
    >
      <div
        style="border-radius: 50px; overflow: hidden; padding: 0px; margin: 0px; display: inline-block; background: rgb(249, 108, 1);"
      >
        <svg
          height="NaN"
          width="NaN"
          x="0"
          y="0"
        >
          <rect
            fill="#F2CA02"
            height="NaN"
            transform="translate(NaN NaN) rotate(529.4 NaN NaN)"
            width="NaN"
            x="0"
            y="0"
          />
          <rect
            fill="#FAA200"
            height="NaN"
            transform="translate(NaN NaN) rotate(336.9 NaN NaN)"
            width="NaN"
            x="0"
            y="0"
          />
          <rect
            fill="#C8142C"
            height="NaN"
            transform="translate(NaN NaN) rotate(276.6 NaN NaN)"
            width="NaN"
            x="0"
            y="0"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
